integration/tests/basic/erc/test_EIP1559.py

```diff
@@ -0,0 +1,1317 @@
+    total_fee_paid = gas_used * effective_gas_price
+
+    # Validate that sender's balance decreased by at least the gas fee
+    assert balance_before - balance_after >= total_fee_paid, "Sender balance did not decrease by gas fee"
+
+    # Verify that the effective gas price does not exceed the max fee per gas
+    assert effective_gas_price <= max_fee_per_gas, (
+        f"Effective gas price: {effective_gas_price}, Max fee per gas: {max_fee_per_gas}"
+    )
+
+    # Validate gas used does not exceed the estimated gas
+    assert gas_used <= estimated_gas, (
+        f"Gas used: {gas_used}, Estimated gas: {estimated_gas}"
+    )
+
+    # Validate cumulative gas used does not exceed block gas limit
+    assert cumulative_gas_used <= block_gas_limit, (
+        f"Cumulative gas used: {cumulative_gas_used}, Block gas limit: {block_gas_limit}"
+    )
+
+
+@allure.feature("EIP Verifications")
+@allure.story("EIP-1559: New Transaction Type Support in Neon")
+@pytest.mark.usefixtures("accounts", "web3_client")
+class TestSendRawTransaction:
+    web3_client: NeonChainWeb3Client
+    accounts: EthAccounts
+
+    def test_transfer_positive(self):
+        validate_transfer_positive(
+            accounts=self.accounts,
+            web3_client=self.web3_client,
+            access_list=None,
+        )
+
+    @pytest.mark.neon_only
+    def test_transfer_without_chain_id_negative(
+            self,
+            json_rpc_client: JsonRPCSession,
+    ):
+        sender = self.accounts[0]
+        recipient = self.web3_client.create_account()
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id=None,
+            from_=sender.address,
+            to=recipient.address,
+            value=10000,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas="auto",
+            max_fee_per_gas="auto",
+            data=None,
+            access_list=None,
+        )
+
+        signed_tx = self.web3_client._web3.eth.account.sign_transaction(tx_params, sender.key)
+        response = json_rpc_client.send_rpc(method="eth_sendRawTransaction", params=signed_tx.rawTransaction.hex())
+        assert "result" not in response
+        assert "error" in response
+        assert response["error"]["code"] == -32000
+
+    def test_deploy_positive(self):
+        validate_deploy_positive(
+            accounts=self.accounts,
+            web3_client=self.web3_client,
+            access_list=None,
+        )
+
+    def test_contract_function_call_positive(
+            self,
+    ):
+        account = self.accounts[0]
+        contract_a, _ = self.web3_client.deploy_and_get_contract(
+            contract="EIPs/EIP161/contract_a_function.sol",
+            version="0.8.12",
+            account=account,
+            contract_name="ContractA",
+        )
+
+        data = contract_a.functions.deploy_contract().build_transaction()["data"]
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
```

Comment:
why can't we use web3 library how it was intended?
doing .build_transaction() here you already has all the parameters and then you calculate them again ignoring what web3 did
maybe we can use contract_a.functions.deploy_contract().build_transaction() whith sending what exactly we want to change (like 2 new parameters)

Just suggestion

Reply:
changed to
```
        tx_params = contract_a.functions.deploy_contract().build_transaction()
        tx_params["nonce"] = self.web3_client.get_nonce(account)
```